daemon: Rename host-reachable services to socket LB (#20369)
cilium-team/cilium: Documentation/gettingstarted/kubeproxy-free.rst

Socket LoadBalancer Bypass in Pod Namespace
*******************************************

> The socket-level loadbalancer acts transparent to Cilium's lower layer datapath
> in that upon connect system call (TCP, connected UDP) or sendmsg as well
> as recvmsg (UDP) the destination IP is checked for an existing service IP
> and one of the service backends is selected as a target, meaning, while
> the application is assuming its connection to the service address, the
> corresponding kernel's socket is actually connected to the backend address
> and therefore no additional lower layer NAT is required.

Reviewer: I find this sentence long and complex to read. Some suggestions:

```suggestion
The socket-level loadbalancer acts transparent to Cilium's lower layer datapath
in that upon ``connect`` (TCP, connected UDP), ``sendmsg`` (UDP), or
``recvmsg`` (UDP) system calls, the destination IP is checked for an existing service IP
and one of the service backends is selected as a target. This means that although
the application assumes it is connected to the service address, the
corresponding kernel socket is actually connected to the backend address
and therefore no additional lower layer NAT is required.
```

> therefore no additional lower layer NAT is required.

Reviewer 2: This is a very succinct and understandable explanation of socket LB, thanks for adding this paragraph!
